### PR TITLE
add node page tests

### PR DESCRIPTION
**Pull request by A-Harby**
Add new test page for farm node includes test cases for:
- Check all twin nodes are listed correctly and visible.
- Search node functionality works probably.  (Function will be commented until the problem is resolved. https://github.com/threefoldtech/tfgrid-sdk-ts/issues/64)
- Sort working fine using all nodes headers. (Function will be commented until the problem is resolved. https://github.com/threefoldtech/tfgrid-sdk-go/issues/53)
- All nodes details are displayed accurately correct.
- Validation for (IPV4, IPV4 Gateway, IPV6, IPV6 Gateway, Domain).
- Add public config to your farm node.
- Remove public config to your farm node.

Also add 2 grid proxy function (To get all twin nodes - To get Node IP config).
And updated space indentation in (test_dedicate, test_homepage, test_swap, test_twin).

### CODE REVIEW

File: tests/frontend_selenium/tests/test_node.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      Result: User should be navigated to Farm Nodes page and see all his node.
+    """
+    node_page, _, grid_proxy, _ = before_test_setup(browser)
+    assert 'Your Farm Nodes' in browser.page_source
+    nodes = grid_proxy.get_twin_node(str(node_page.twin_id))
+    for node in nodes:
+        assert str(node['nodeId']) in browser.page_source
+
+
+# def test_search_node(browser):
```
Comment (mohamedamer453) [inline tests/frontend_selenium/tests/test_node.py:39]: For the commented tests, please add a comment above each one mentioning that the these tests will remain commented until the following issues are resolved:

- Search 
   https://github.com/threefoldtech/tfgrid_dashboard/issues/392

- Sort
   https://github.com/threefoldtech/tfgridclient_proxy/issues/204

File: tests/frontend_selenium/tests/test_node.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+          - Click on Farm from side menu.
+          - Click on 'Add a public config' button.
+          - Enter values and click on save button.
+      Result: User should be able to add the public config to his farm node.
+    """
+    node_page, polka_page, grid_proxy, password = before_test_setup(browser)
+    nodes = grid_proxy.get_twin_node(str(node_page.twin_id))
+    rand_node = random.randint(0,len(nodes)-1)
+    node_id = nodes[rand_node]['nodeId']
+    old_ipv4 = nodes[rand_node]['publicConfig']['gw4']
```
Comment (mohamedamer453) [inline tests/frontend_selenium/tests/test_node.py]: In this section, is there a specific reason for comparing the old gateway ip to the generated ipv4? shouldn't the old_ipv4 be `old_ipv4 = nodes[rand_node]['publicConfig']['ipv4']`

**Review by mohamedamer453 — CHANGES_REQUESTED** [review]

File: tests/frontend_selenium/tests/test_node.py (hunk 1 above)
Comment (A-Harby) [reply tests/frontend_selenium/tests/test_node.py:39]: Description updated.

File: tests/frontend_selenium/tests/test_node.py (hunk 2 above)
Comment (A-Harby) [reply tests/frontend_selenium/tests/test_node.py]: It worked as function was just waiting for any changes in the grid.
But your comment was right and updated it in [faaca5c](https://github.com/threefoldtech/tfgrid_dashboard/pull/355/commits/faaca5c57838256b77976f3d03055ab8594f64b8)

**Review by mohamedamer453 — APPROVED** [review]
Requested changes were resolved and the docs were updated with the new variable.

**Pull request merged**